src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -191,3 +204,54 @@ func (q *qemuAmd64) appendImage(ctx context.Context, devices []govmmQemu.Device,
 func (q *qemuAmd64) appendBridges(devices []govmmQemu.Device) []govmmQemu.Device {
 	return genericAppendBridges(devices, q.Bridges, q.qemuMachine.Type)
 }
+
+// enable protection
+func (q *qemuAmd64) enableProtection() error {
+	flags, err := CPUFlags(procCPUInfo)
+	if err != nil {
+		return err
+	}
+
+	// TDX is supported and properly loaded when the firmware directory exists or `tdx` is part of the CPU flags
+	if d, err := os.Stat(tdxSysFirmwareDir); (err != nil && d.IsDir()) || flags[tdxCPUFlag] {
+		q.protection = tdxProtection
+		if q.qemuMachine.Options != "" {
+			q.qemuMachine.Options += ","
+		}
+		q.qemuMachine.Options += "kvm-type=tdx,confidential-guest-support=tdx"
+		q.kernelParams = append(q.kernelParams, Param{"tdx_guest", ""})
+		virtLog.WithFields(logrus.Fields{
+			"subsystem":     "qemuAmd64",
+			"machine":       q.qemuMachine,
+			"kernel-params": q.kernelParameters}).
+			Info("Enabling TDX guest protection")
+		return nil
+	}
+
+	// TODO: Add support for other x86_64 technologies: SEV
+
+	return fmt.Errorf("This system doesn't support Confidential Computing (Guest Protection)")
+}
+
+// add protection device
+func (q *qemuAmd64) addProtectionDevice(devices []govmmQemu.Device, firmware string) ([]govmmQemu.Device, bool, error) {
+	switch q.protection {
+	case tdxProtection:
+		id := q.devLoadersCount
+		q.devLoadersCount += 1
+		return append(devices,
+			govmmQemu.Object{
+				Driver:   govmmQemu.Loader,
+				Type:     govmmQemu.TDXGuest,
+				ID:       "tdx",
+				DeviceID: fmt.Sprintf("fd%d", id),
+				Debug:    false,
+				File:     firmware,
+			}), false, nil
+	case noneProtection:
+		return devices, false, fmt.Errorf("Could not add protection device: protection is none")
```

Comment:
I don't think this should be an error case. If `noneProtection`, then this function is a NO-OP, and just returns `devices, bios, nil`.

As a matter of fact, I believe the `appendProtectionDevice` from `qemu.go:createSandbox()` should just be:

```Go
qemuConfig.Devices, useBios, err = q.arch.addProtectionDevice(qemuConfig.Devices, firmwarePath)
if err != nil {
	return err
}
```

and should not depend on the `q.config.guestProtection` toggle.